packages/plugin-ext-vscode/src/node/plugin-vscode-directory-handler.ts

```diff
@@ -49,7 +49,17 @@ export class PluginVsCodeDirectoryHandler implements PluginDeployerDirectoryHand
 
     async handle(context: PluginDeployerDirectoryHandlerContext): Promise<void> {
         await this.copyDirectory(context);
-        context.pluginEntry().accept(PluginDeployerEntryType.BACKEND);
+        const types: PluginDeployerEntryType[] = [];
+        const packageJson: PluginPackage = context.pluginEntry().getValue('package.json');
+        if (packageJson.main) {
+            types.push(PluginDeployerEntryType.BACKEND);
+        }
+        if (packageJson.browser) {
+            types.push(PluginDeployerEntryType.FRONTEND);
+        }
+
+        context.pluginEntry().accept(...types);
+        return Promise.resolve();
```

Comment:
Unnecessary white space and return value:

```suggestion
        context.pluginEntry().accept(...types);
```

Reply:
Changed.